lib/ramble/ramble/software_environments.py

```diff
@@ -481,7 +548,12 @@ def unused_packages(self):
             (TemplatePackage) Each unused template package in this group
         """
         for pkg in self._package_templates.values():
```

Comment:
Looks like this relies on the equivalence check between TemplatePackage and the rendered SoftwarePackage. I think this would be trickier to do now that the template does not have the concrete specs. Is it not sufficient to simply check for the keys?

Reply:
I'm not sure if this comment is related to the line it's anchored to. This looks like it just checks if the template package rendered something or not.